### ISSUE TITLE
Qlot install hangs on directories with a large number of subdirs

### ISSUE THREAD

**Issue by theneosloth** [fukamachi/qlot#232]
## Describe the bug
I use direnv with flake.nix in order to manage my development dependencies. What that ends up doing is creating a `.direnv` directory in my project that contains a link to a _very_ large directory that does not necessarily contain project related files.

Running any qlot command in this project will cause the command to fail because it keeps recursively traversing that path, even though it is not a part of my project's files

## Reproducible steps
Run qlot in a directory containing very large number of subdirs and files
Watch it hang
Ctrl+C to get the current backtrace
## Error messages
```
0: ("bogus stack frame")
1: (SB-UNIX:UNIX-STAT "/$PROJECT_DIR/.direnv/flake-inputs/j4zjf3ph0lr39sg02zl0mk4vpf059r0z-source/pkgs/applications/misc/cmatrix/default.nix")
2: (SB-IMPL::%QUERY-FILE-SYSTEM #P"/$PROJECT_DIR/.direnv/flake-inputs/j4zjf3ph0lr39sg02zl0mk4vpf059r0z-source/pkgs/applications/misc/cmatrix/default.nix" :EXISTEN)
3: ((LABELS SB-IMPL::RECORD :IN DIRECTORY) #P"/$PROJECT_DIR/.direnv/flake-inputs/j4zjf3ph0lr39sg02zl0mk4vpf059r0z-source/pkgs/applications/misc/cmatrix/default.ni)
4: ((FLET SB-IMPL::ITERATE :IN SB-EXT:MAP-DIRECTORY) #<FUNCTION (FLET SB-IMPL::ONE-ITER :IN SB-IMPL::CALL-WITH-NATIVE-DIRECTORY-ITERATOR) {7F08592EDB9B}>)
5: (SB-IMPL::CALL-WITH-NATIVE-DIRECTORY-ITERATOR #<FUNCTION (FLET SB-IMPL::ITERATE :IN SB-EXT:MAP-DIRECTORY) {7F08592EDC2B}> "/$PROJECT_DIR/.direnv/flake-inputs/j)
6: (SB-EXT:MAP-DIRECTORY #<FUNCTION (LAMBDA (SB-IMPL::FILE) :IN SB-IMPL::MAP-MATCHING-ENTRIES) {1003F4477B}> #P"/$PROJECT_DIR/.direnv/flake-inputs/j4zjf3ph0lr39sg)
7: (SB-IMPL::MAP-MATCHING-DIRECTORIES #<FUNCTION (LAMBDA #1=(DIRECTORY) :IN . #1#) {1003F4473B}> #P"/$PROJECT_DIR/.direnv/flake-inputs/j4zjf3ph0lr39sg02zl0mk4vpf0)
8: (DIRECTORY #P"/$PROJECT_DIR/.direnv/flake-inputs/j4zjf3ph0lr39sg02zl0mk4vpf059r0z-source/pkgs/applications/misc/cmatrix/*.*" :RESOLVE-SYMLINKS NIL)
9: (UIOP/FILESYSTEM:SUBDIRECTORIES #P"/$PROJECT_DIR/.direnv/flake-inputs/j4zjf3ph0lr39sg02zl0mk4vpf059r0z-source/pkgs/applications/misc/cmatrix/")
10: (UIOP/FILESYSTEM:COLLECT-SUB*DIRECTORIES #P"/$PROJECT_DIR/.direnv/flake-inputs/j4zjf3ph0lr39sg02zl0mk4vpf059r0z-source/pkgs/applications/misc/cmatrix/" #<FUNC)
11: (UIOP/FILESYSTEM:COLLECT-SUB*DIRECTORIES #P"/$PROJECT_DIR/.direnv/flake-inputs/j4zjf3ph0lr39sg02zl0mk4vpf059r0z-source/pkgs/applications/misc/" #<FUNCTION (FL)
12: (UIOP/FILESYSTEM:COLLECT-SUB*DIRECTORIES #P"/$PROJECT_DIR/.direnv/flake-inputs/j4zjf3ph0lr39sg02zl0mk4vpf059r0z-source/pkgs/applications/" #<FUNCTION (FLET AS)
13: (UIOP/FILESYSTEM:COLLECT-SUB*DIRECTORIES #P"/$PROJECT_DIR/.direnv/flake-inputs/j4zjf3ph0lr39sg02zl0mk4vpf059r0z-source/pkgs/" #<FUNCTION (FLET ASDF/SOURCE-REG)
14: (UIOP/FILESYSTEM:COLLECT-SUB*DIRECTORIES #P"/$PROJECT_DIR/.direnv/flake-inputs/j4zjf3ph0lr39sg02zl0mk4vpf059r0z-source/" #<FUNCTION (FLET ASDF/SOURCE-REGISTRY)
15: (UIOP/FILESYSTEM:COLLECT-SUB*DIRECTORIES #P"/$PROJECT_DIR/.direnv/flake-inputs/" #<FUNCTION (FLET ASDF/SOURCE-REGISTRY::COLLECTP :IN ASDF/SOURCE-REGISTRY:COLL)
16: (UIOP/FILESYSTEM:COLLECT-SUB*DIRECTORIES #P"/$PROJECT_DIR/.direnv/" #<FUNCTION (FLET ASDF/SOURCE-REGISTRY::COLLECTP :IN ASDF/SOURCE-REGISTRY:COLLECT-SUB*DIREC)
17: (UIOP/FILESYSTEM:COLLECT-SUB*DIRECTORIES #P"/$PROJECT_DIR/" #<FUNCTION (FLET ASDF/SOURCE-REGISTRY::COLLECTP :IN ASDF/SOURCE-REGISTRY:COLLECT-SUB*DIRECTORIES-A)
18: (ASDF/SOURCE-REGISTRY:COMPUTE-SOURCE-REGISTRY (:SOURCE-REGISTRY :IGNORE-INHERITED-CONFIGURATION (:ALSO-EXCLUDE ".qlot") (:DIRECTORY #P"/home/neosloth/.roswell/local-projects/fukamachi/qlot/") (:TREE #P"/m)
19: (SB-INT:SIMPLE-EVAL-IN-LEXENV (SETUP-SOURCE-REGISTRY) #<NULL-LEXENV>)
20: (SB-EXT:EVAL-TLF (SETUP-SOURCE-REGISTRY) 4 NIL)
21: ((LABELS SB-FASL::EVAL-FORM :IN SB-INT:LOAD-AS-SOURCE) (SETUP-SOURCE-REGISTRY) 4)
22: ((LAMBDA (SB-KERNEL:FORM &KEY :CURRENT-INDEX &ALLOW-OTHER-KEYS) :IN SB-INT:LOAD-AS-SOURCE) (SETUP-SOURCE-REGISTRY) :CURRENT-INDEX 4)
23: (SB-C::%DO-FORMS-FROM-INFO #<FUNCTION (LAMBDA (SB-KERNEL:FORM &KEY :CURRENT-INDEX &ALLOW-OTHER-KEYS) :IN SB-INT:LOAD-AS-SOURCE) {7F08592EE6BB}> #<SB-C::SOURCE-INFO {1003F766F3}> SB-C::INPUT-ERROR-IN-LOAD)
24: (SB-INT:LOAD-AS-SOURCE #<SB-INT:FORM-TRACKING-STREAM for "file /$PROJECT_DIR/.qlot/local-init/qlot-99-setup.lisp" {1003F9C533}> :VERBOSE NIL :PRINT NIL :CONTE)
25: ((LABELS SB-FASL::LOAD-STREAM-1 :IN LOAD) #<SB-INT:FORM-TRACKING-STREAM for "file /$PROJECT_DIR/.qlot/local-init/qlot-99-setup.lisp" {1003F9C533}> NIL)
26: (SB-FASL::CALL-WITH-LOAD-BINDINGS #<FUNCTION (LABELS SB-FASL::LOAD-STREAM-1 :IN LOAD) {7F08592EEA7B}> #<SB-INT:FORM-TRACKING-STREAM for "file /$PROJECT_DIR/.q)
27: (LOAD #P"/$PROJECT_DIR/.qlot/local-init/qlot-99-setup.lisp" :VERBOSE NIL :PRINT NIL :IF-DOES-NOT-EXIST :ERROR :EXTERNAL-FORMAT :DEFAULT)
28: (QUICKLISP-CLIENT:SETUP)
```
## Versions:
 - Qlot version: 1.5.0
 - Where did you get it from: github through roswell
 - OS: Ubuntu 22.04.3
 - Lisp: SBCL 2.4.1

Is there a way to add more directories to the ASDF's ignored registries? I'm seeing it excludes qlot under
`(ASDF/SOURCE-REGISTRY:COMPUTE-SOURCE-REGISTRY (:SOURCE-REGISTRY :IGNORE-INHERITED-CONFIGURATION (:ALSO-EXCLUDE .qlot") (:DIRECTORY #P"PROJECT_PATH/m)`?

Thank you!

**Comment by fukamachi** [comment]
Thank you for pointing out!
Qlot doesn't provide such a functionality, but adding .direnv to `asdf:*default-source-registry-exclusions*` will possibly avoid it.

**Comment by fukamachi** [comment]
I don't have an idea to allow specific directory names to be ignored, but I added `.direnv` to the source-registry.conf in #233 for now.
Any better solutions are welcomed.

**Comment by theneosloth** [comment]
Great thank you `asdf:*default-source-registry-exclusions*` seems like the clean solution here if this comes up again. The readme mentions that roswell ignores any directory prefixed with a dot, are there concerns with having qlot work the same way?

Might be too heavy handed though

**Comment by fukamachi** [comment]
Fixed by #233.
